manifests/database.pp

```diff
@@ -26,6 +26,11 @@
       value => false,
       dry   => true,
     } ~>
+    exec { 'restart-services':
+      command => '/usr/bin/katello-service restart &> /dev/null && /usr/sbin/service-wait foreman-tasks restart &> /dev/null ',
+      timeout     => 1800,
+      require => Package['katello-service'],
+    } ~>
```

Comment:
A few things:

1. Execs have to be gated by something, you can control the run by subscribing to another resource, or monitoring a particular file's creation. You can't have an exec that always runs which this does. https://docs.puppet.com/puppet/latest/reference/types/exec.html

2. katello-service can't be a requirement inside of Foreman, this is restarting services and executing commands that are totally irrelevant to Foreman

I think you'll need to solve this inside a hook or something, what was driving the requirement for this?